### DISCUSSION TITLE
Use Blocks to call method on an object

### DISCUSSION THREAD

**Discussion by ggramlich** [pybricks/support#1354] [Q&A] [answered]
With Pybricks Block Coding Is it possible to call an arbitrary method on an object, which I have imported or which is setup as a device (e.g. the drivebase object itself)?
I know, how to use the import and external tasks blocks and understand, how they are translated to python. But I don't see a direct way to make a method call.
This is not a feature request. I am just curious, if it is possible.

**Comment by laurensvalk** [comment, marked as answer]
It currently isn't possible out of the box.

It is another one of those things that probably could be done, but we intentionally left it out to start with.

Similar to the inline functions you asked about, I think we can consider it though - combined, it gets pretty flexible for power users that way.

This was also requested by a third party accessory maker - they wanted to make a class in Python that the user could instantiate and use with blocks just like a regular device.

If we're doing this, we may want to take it a step further and allow providing an actual block specification so teachers can make their own block palette. 

Even if this sounds somewhat complex, it's still way easier than it used to be with NXT and EV3, where you needed a custom toolkit from NI just to make a few blocks.
